modules/nf-core/fastqc/main.nf

```diff
@@ -21,9 +21,14 @@ process FASTQC {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def old_new_pairs = reads instanceof Path || reads.size() == 1 ? [[ reads, "${prefix}.${reads.extension}" ]] : reads.withIndex().collect { entry, index -> [ entry, "${prefix}_${index + 1}.${entry.extension}" ] }
```

Comment:
Looks good to me and kudos to the Groovy magic :bowtie:. But now much more complex to track this logic overall that could be handled by having an additional input value to the process?

Reply:
I think going back to the old way introduces much more input and cases, and duplicated code making it less maintainable. There's now tests for multiple fastx and bam too. But then again, if someone trying to update can't read the Groovy magic, that doesn't help either.